classes/autoptimizeStyles.php

```diff
@@ -351,7 +354,7 @@ public function read( $options )
                         $new_tag = $this->optionally_defer_excluded( $new_tag, $url );
 
                         // And replace!
-                        if ( '' !== $new_tag && $new_tag !== $tag ) {
+                        if ( '' !== $tag && $new_tag !== $tag ) {
```

Comment:
I think this has to remain `$new_tag` Sjoerd.

Reply:
If $new_tag remains then the style tag will not be removed for the empty cache file and the original NOT minified link will still be used. May be you can think of a other ways to do this? It needs to be done here because the link needs to be removed.

Reply:
but `$tag` will never by empty, so that condition would always be `true`?  In order to remove the old tag `$tag` (which I'm not too sure it AO's responsibility tbh, but well, if we can why not) you could probably simply do;

```php
if ( $new_tag !== $tag ) {
    $this->content = str_replace( $tag, $new_tag, $this->content );
}
```

Reply:
Yes this a perfect solution! I will change this.